From http://xenbits.xen.org/hg/linux-2.6.18-xen.hg/rev/05b894bd0a0f
From: xen-devel@lists.xenproject.org
Patch-mainline: Never, SUSE-Xen specific
Subject: xenbus: inspect the correct type in xenbus_dev_request_and_reply()

Looking at the local variable is wrong for the XS_TRANSACTION_END case:
In case of an earlier error transaction_end() must not be called.

Signed-off-by: Jan Beulich <jbeulich@suse.com>

--- a/drivers/xen/xenbus/xenbus_xs.c
+++ b/drivers/xen/xenbus/xenbus_xs.c
@@ -220,7 +220,7 @@ void *xenbus_dev_request_and_reply(struc
 
 	mutex_unlock(&xs_state.request_mutex);
 
-	if ((type == XS_TRANSACTION_END) ||
+	if ((msg->type == XS_TRANSACTION_END) ||
 	    ((type == XS_TRANSACTION_START) && (msg->type == XS_ERROR)))
 		transaction_end();
 
